lib/galaxy/datatypes/molecules.py

```diff
@@ -674,6 +795,684 @@ def set_peek(self, dataset):
             dataset.blurb = 'file purged from disk'
 
 
+class Cell(GenericMolFile):
+    """
+    CASTEP CELL format.
+    """
+
+    file_ext = "cell"
+    MetadataElement(
+        name="atom_data",
+        default=[],
+        desc="Atom symbols and positions",
+        readonly=True,
+        visible=False,
+    )
+    MetadataElement(
+        name="number_of_atoms",
+        desc="Number of atoms",
+        readonly=True,
+        visible=True,
+    )
+    MetadataElement(
+        name="chemical_formula",
+        desc="Chemical formula",
+        readonly=True,
+        visible=True,
+    )
+    MetadataElement(
+        name="is_periodic",
+        desc="Periodic boundary conditions",
+        readonly=True,
+        visible=True,
+    )
+    MetadataElement(
+        name="lattice_parameters",
+        desc="Lattice parameters",
+        readonly=True,
+        visible=True,
+    )
+
+    def sniff(self, filename):
+        """
+        Try to guess if the file is a CASTEP CELL file.
+
+        A fingerprint for CELL files is the use of %BLOCK and %ENDBLOCK to
+        denote data blocks (not case sensitive).
+
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname('Si_uppercase.cell')
+        >>> Cell().sniff(fname)
+        True
+        >>> fname = get_test_fname('Si_lowercase.cell')
+        >>> Cell().sniff(fname)
+        True
+        >>> fname = get_test_fname('Si.cif')
+        >>> Cell().sniff(fname)
+        False
+        """
+        with open(filename) as f:
+            cell = f.read(1000)
+        if re.search(r"\n%BLOCK", cell, flags=re.IGNORECASE) is not None:
+            if re.search(r"\n%ENDBLOCK", cell, flags=re.IGNORECASE) is not None:
+                return True
+        else:
+            return False
+
+        # if %BLOCK was found but not %ENDBLOCK, check the rest of the file
+        with open(filename) as f:
+            cell = f.read()
+        if re.search(r"\n%ENDBLOCK", cell, flags=re.IGNORECASE) is not None:
+            return True
+        else:
+            return False
```

Comment:
FilePrefix keeps the first MegaBytes of the file to sniff in memory. If you're expecting to find a full block in the first megabyte this would be a more efficient alternative:

```suggestion
    def sniff_prefix(self, file_prefix: FilePrefix):
        """
        Try to guess if the file is a CASTEP CELL file.

        A fingerprint for CELL files is the use of %BLOCK and %ENDBLOCK to
        denote data blocks (not case sensitive).

        >>> from galaxy.datatypes.sniff import get_test_fname
        >>> fname = get_test_fname('Si_uppercase.cell')
        >>> Cell().sniff(fname)
        True
        >>> fname = get_test_fname('Si_lowercase.cell')
        >>> Cell().sniff(fname)
        True
        >>> fname = get_test_fname('Si.cif')
        >>> Cell().sniff(fname)
        False
        """
        start_found = False
        for line in file_prefix.line_iterator():
            if not start_found and line.lower().startswith('%block'):
                start_found = True
            elif start_found and line.lower().startswith('%endblock'):
                return True
        return False
```